Several TPM fixes

herve-elter opened the pull request:
Please find some modifications in order to have the TPM2.0 working on my custom boards based on Intel Haracuvar (Denverton SoC) with a TPM2.0 Infineon SLB9665 and Bios Coreboot 4.19

May be it can help!
Regards

Commit:
Fix TPM Dectetion issues
Prevents variable TPM20Present and TPM12Present from being set simultaneously
to avoid Tcg2Dxe init failure

Commit:
Fix TPM2 Bios Setup Hang
Avoid Hang in Bios Setup->Device Manager->TCG2 Configuration: Error Get TPM Manufacturer

Commit:
Fix TPM PPI issues: Add Flush_PPI_Cache function
With some Intel processors (Intel Denverton) it is necessary to flush the cache before rebooting otherwise mppi data are lost

Commit:
TPM2 update default var

MrChromebox commented on MdeModulePkg/Logo/Logo.bmp:
not sure we want this as part of the PR :)

herve-elter replied:
Oups yes sorry!

MrChromebox replied:
you can just rebase the PR and drop the commit, then force-push the branch. No need for a revert

Commit:
Secureboot: fix dbx unable to be updated by OS
According to https://uefi.org/revocationlistfile dbx variable should be stored with NV+BS+RT+AT flags

Commit:
Secureboot: dbx update to latest database

MrChromebox commented:
@herve-elter do you have a TPM 1.2 system to test as well? I'm not having any luck here with an older Chromebox w/Infineon TPM 1.2

herve-elter commented:
A quick test with a TPM1.2 SLB9660:

Under Coreboot:
Found TPM SLB9660 TT 1.2 by Infineon
TPM: setup succeeded
coreboot TPM 1.2 measurements:

 PCR-2 d624e022fd26262640adef6b7a6a08fdb0e1ba13 SHA1 [FMAP: FMAP]
 PCR-2 c531c25b0acfa87661cab3ce50038517e6ddf4c1 SHA1 [CBFS: bootblock]
 PCR-2 d822efe4301394f6d9a009a3ae7b480815d008d5 SHA1 [CBFS: fallback/romstage]
 PCR-2 080df1eb523c8b873ebec1d99d3dcf9afb3373a9 SHA1 [CBFS: fspm.bin]
 PCR-2 b9b68ad4b3d5d684c270ceb220b15b60106eb6a3 SHA1 [CBFS: fallback/postcar]
 PCR-2 c951102a7d9b55f6c82e013668ffc16c8b58f2b0 SHA1 [CBFS: fallback/ramstage]
 PCR-2 6fe1251f43855bfefc3df36c396d1e51261ddeaf SHA1 [CBFS: fsps.bin]
 PCR-2 da46ea3da13da202b4b30a80f68f8d3e7bd1f31c SHA1 [CBFS: cpu_microcode_blob.bin]
 PCR-2 705110757c59dd715aefce61fb12c442bfc977a3 SHA1 [CBFS: fallback/dsdt.aml]
 PCR-2 89cc598bf329d0f8aeed02dd4cad2a31ff35911b SHA1 [CBFS: fallback/payload]

Under edk2:
TPM 2.0 present 0
TPM 1.2 present 1

Loading driver at 0x0007EEF4000 EntryPoint=0x0007EEF75E9 TcgDxe.efi
InstallProtocolInterface: BC62157E-3E33-4FEC-9920-2D3B36D750DF 7EF2C618
ProtectUefiImageCommon - 0x7EF2C7C0
  - 0x000000007EEF4000 - 0x0000000000005A80
PROGRESS CODE: V03040002 I0

Loading driver at 0x0007DF98000 EntryPoint=0x0007DF9FA04 Tcg2Dxe.efi
InstallProtocolInterface: BC62157E-3E33-4FEC-9920-2D3B36D750DF 7EEFAD98
ProtectUefiImageCommon - 0x7EEFAAC0
  - 0x000000007DF98000 - 0x0000000000011D40
PROGRESS CODE: V03040002 I0
WARNING: Tpm2RegisterTpm2DeviceLib - does not support 286BF25A-C2C3-408C-B3B4-25E6758B7317 registration
No TPM2 instance required!
Error: Image at 0007DF98000 start failed: Unsupported
PROGRESS CODE: V03040003 I0

But nothing's else.. I have to had more traces...

MrChromebox commented:
@herve-elter not getting a TPM menu here on ChromeOS devices with CR50/TPM2, see anything in the log?

https://dpaste.org/XUFn0

herve-elter commented:
@MrChromebox  yes line 1606
Loading driver at 0x0009923E000 EntryPoint=0x0009924596D Tcg2Dxe.efi
InstallProtocolInterface: BC62157E-3E33-4FEC-9920-2D3B36D750DF 993C9998
ProtectUefiImageCommon - 0x993C94C0
  - 0x000000009923E000 - 0x0000000000011DC0
PROGRESS CODE: V03040002 I0
TPM2 not detected!

MrChromebox commented:
@herve-elter any idea why? It it only supporting specific TPM chips?

herve-elter commented:
@MrChromebox try to enable DEBUG_VERBOSE trace.
In UeufiPayloadPkg.dsc set  gEfiMdePkgTokenSpaceGuid.PcdDebugPrintErrorLevel|0x80**4**0004F

End of work for me today. Good weekend I will look at this next week
